Sources/PointFree/Account/PrivateRss.swift

```diff
@@ -0,0 +1,255 @@
+import Optics
+import Prelude
+import Tuple
+
+let accountRssMiddleware: Middleware<StatusLineOpen, ResponseEnded, Tuple2<Database.User.Id, Database.User.RssSalt>, Data> =
+{ fetchUser >=> $0 }
+  <<< filterMap(require1 >>> pure, or: invalidatedFeedMiddleware)
+  <<< validateUserAndSalt
+  <<< fetchSubscription
+  <<< filterMap(validateActiveSubscriber, or: invalidatedFeedMiddleware)
+  <| map(lower)
+  >>> writeStatus(.ok)
+  >=> trackFeedRequest
+  >=> respond(privateEpisodesFeedView, contentType: .text(.init("xml"), charset: .utf8))
+  >=> clearHeadBody
+
+private func invalidatedFeedMiddleware<A>(_ conn: Conn<StatusLineOpen, A>) -> IO<Conn<ResponseEnded, Data>> {
+  return conn.map(const(unit))
+    |> writeStatus(.ok)
+    >=> respond(invalidatedFeedView, contentType: .text(.init("xml"), charset: .utf8))
+    >=> clearHeadBody
+}
+
+private func validateActiveSubscriber<Z>(
+  data: T3<Database.Subscription?, Database.User, Z>
+  ) -> IO<T2<Database.User, Z>?> {
+
+  return IO {
+    guard let subscription = get1(data) else { return nil }
+    let user = get2(data)
+
+    return SubscriberState(user: user, subscription: subscription).isActive
+      ? user .*. rest(data)
+      : nil
+  }
+}
+
+private func validateUserAndSalt<Z>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, Z>, Data>)
+  -> Middleware<StatusLineOpen, ResponseEnded, T3<Database.User, Database.User.RssSalt, Z>, Data> {
+
+    return { conn in
+      guard get1(conn.data).rssSalt == get2(conn.data) else {
+        return conn
+          |> invalidatedFeedMiddleware
+      }
+      return conn.map(const(get1(conn.data) .*. rest(conn.data)))
+        |> middleware
+    }
+}
+
+private func trackFeedRequest<I>(_ conn: Conn<I, Database.User>) -> IO<Conn<I, Database.User>> {
+
+  return Current.database.createFeedRequestEvent(
+    .privateEpisodesFeed,
+    conn.request.allHTTPHeaderFields?["User-Agent"] ?? "",
+    conn.data.id
+    )
+    .withExcept(notifyError(subject: "Create Feed Request Event Failed"))
+    .run
+    .map { _ in conn }
+}
+
+private let privateEpisodesFeedView = itunesRssFeedLayout <| View<Database.User> { user -> Node in
+  node(
+    rssChannel: privateRssChannel(user: user),
+    items: items(forUser: user)
+  )
+}
+
+func privateRssChannel(user: Database.User) -> RssChannel {
+  let description = """
+Point-Free is a video series about functional programming and the Swift programming language. Each episode
+covers a topic that may seem complex and academic at first, but turns out to be quite simple. At the end of
+each episode we’ll ask “what’s the point?!”, so that we can bring the concepts back down to earth and show
+how these ideas can improve the quality of your code today.
+
+---
+
+This is a private feed associated with the Point-Free account \(user.email). Please do not share this link
+with anyone else.
+"""
+  let title = "Point-Free Videos (Private feed for \(user.email.rawValue))"
```

Comment:
title of their feed with have their email embedded in it to remind them it's private